### ISSUE TITLE
dotnet pack does not support arbitrary configurations

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/cli#4488]
PJ allowed passing in arbitrary configurations, and would output to the appropriate bin/{Configuration} subdirectory. CSProj fails with:

```
error : The OutputPath property is not set for project 'TestLibraryWithConfiguration.csproj'.  Please check to make sure that you have specified a valid combination of Configuration and Platform for this project.  Configuration='Test'  Platform='AnyCPU'.  You may be seeing this message because you are trying to build a project without a solution file, and have specified a non-default Configuration or Platform that doesn't exist for this project.
```

@nguerrera how do we enable custom configurations?

**Comment by piotrpMSFT** [comment]
@srivatsn do you know what it would take to re-enable this behavior?

**Comment by livarcocc** [comment]
@rohit21agrawal can you please take a look?

**Comment by rohit21agrawal** [comment]
@eerhardt @nguerrera kno  about this issue. this is not related to pack.

**Comment by eerhardt** [comment]
MSBuild has this explicit check in it to make sure you don't pass in invalid configurations.  It is the same check that causes all the "`Configuration` is set as an environment variable, and OutputPath isn't set so restore fails" problems we've been fighting with and needing to pass `/p:SkipInvalidConfigurations=true` into restore.

See

https://github.com/Microsoft/msbuild/blob/bf9b21cc7869b96ea2289ff31f6aaa5e1d525a26/src/XMakeTasks/Microsoft.Common.CurrentVersion.targets#L158-L166

https://github.com/Microsoft/msbuild/blob/bf9b21cc7869b96ea2289ff31f6aaa5e1d525a26/src/XMakeTasks/Microsoft.Common.CurrentVersion.targets#L714-L748

@rainersigwald @cdmihai @jeffkl

**Comment by nguerrera** [comment]
The biggest challenge here is deciding on the settings applied to such a configuration. We can go ahead and let the build proceed to an arbitrary directory. This trades off any enforcement against misspelling of configurations. So I say 'releease' or 'deebug', and get something rather arbitrary instead of an error.

We've talked about this lenient approach and it's the cheapest way to unblock adding a new configuration in VS, which is currently broken, but just letting things proceed wouldn't actually give the right behavior. In VS, when you create a new configuration, you say from which other configuration you want settings to inherit. If we just set the output path and nothing else, you wouldn't inherit appropriately.

So we have to decide two things:

1. Do we want enforcement of configurations or is it garbage-in-garbage-out?

2. Do we support this 'make a new configuration X with settings seeded from Y' experience that has always existed in VS? If so, what's the menchanism for triggering this behavior in a new world where there needn't be a property group for every Configuration|Platform combination? If not, we need to track VS changes.

In short, this isn't as simple as "just always set the output path". There are consequences.

**Comment by nguerrera** [comment]
This is really an SDK issue (though some work in msbuild might be required) not a CLI or nuget issue. It is related to dotnet/sdk#203

**Comment by livarcocc** [comment]
Created a copy of this in the SDK: https://github.com/dotnet/sdk/issues/328
